src/cls/rbd/cls_rbd.cc

```diff
@@ -3177,7 +3177,7 @@ int dir_state_assert(cls_method_context_t hctx, bufferlist *in, bufferlist *out)
     return -EINVAL;
   }
 
-  cls::rbd::DirectoryState on_disk_directory_state;
+  cls::rbd::DirectoryState on_disk_directory_state = directory_state;
```

Comment:
@dillaman is this related to the versioning of C++ API? if not, can we move it into its own commit?

Reply:
Done